tests/api_workflow/test_api_workflow_compute_worker.py

```diff
@@ -557,6 +554,82 @@ def test_download_compute_worker_run_artifacts(mocker: MockerFixture) -> None:
     mock_download_compute_worker_run_artifact.assert_has_calls(calls=calls)
     assert mock_download_compute_worker_run_artifact.call_count == len(calls)
 
+def test_get_compute_worker_run_tags__no_tags(mocker: MockerFixture) -> None:
+    client = ApiWorkflowClient(token="123", dataset_id="dataset-0")
+    mock_compute_worker_api = mocker.create_autospec(DockerApi, spec_set=True).return_value
+    mock_compute_worker_api.get_docker_run_tags.return_value = []
+    client._compute_worker_api = mock_compute_worker_api
+    tags = client.get_compute_worker_run_tags(run_id="run-0")
+    assert len(tags) == 0
+    mock_compute_worker_api.get_docker_run_tags.assert_called_once_with(run_id="run-0")
+
+def test_get_compute_worker_run_tags__single_tag(mocker: MockerFixture) -> None:
+    client = ApiWorkflowClient(token="123", dataset_id="dataset-0")
+    mock_compute_worker_api = mocker.create_autospec(DockerApi, spec_set=True).return_value
+    mock_compute_worker_api.get_docker_run_tags.return_value = [
+        TagData(
+            id="tag-0",
+            dataset_id="dataset-0",
+            prev_tag_id=None,
+            bit_mask_data="0x1",
+            name="tag-0",
+            tot_size=0,
+            created_at=0,
+            changes=dict(),
+            run_id="run-0",
+        )
+    ]
+    client._compute_worker_api = mock_compute_worker_api
+    tags = client.get_compute_worker_run_tags(run_id="run-0")
+    assert len(tags) == 1
+    mock_compute_worker_api.get_docker_run_tags.assert_called_once_with(run_id="run-0")
+
+def test_get_compute_worker_run_tags__multiple_tags(mocker: MockerFixture) -> None:
+    client = ApiWorkflowClient(token="123", dataset_id="dataset-0")
+    mock_compute_worker_api = mocker.create_autospec(DockerApi, spec_set=True).return_value
+    tag_0 = TagData(
+        id="tag-0",
+        dataset_id="dataset-0",
+        prev_tag_id=None,
+        bit_mask_data="0x1",
+        name="tag-0",
+        tot_size=0,
+        created_at=0,
+        changes=dict(),
+        run_id="run-0",
+    )
+    tag_1 = TagData(
+        id="tag-1",
+        dataset_id="dataset-0",
+        prev_tag_id=None,
+        bit_mask_data="0x1",
+        name="tag-1",
+        tot_size=0,
+        created_at=1,
+        changes=dict(),
+        run_id="run-0",
+    )
+    # tag from a different dataset
+    tag_2 = TagData(
+        id="tag-2",
+        dataset_id="dataset-1",
+        prev_tag_id=None,
```

Comment:
id of tag_0 or tag_1

Reply:
I think this one is correct because the `dataset_id` is different.

Reply:
ah true sorry, didn't see that